src/pages/Facility/settings/devices/DevicesList.tsx

```diff
@@ -7,82 +7,154 @@ import CareIcon from "@/CAREUI/icons/CareIcon";
 import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
 
+import Page from "@/components/Common/Page";
 import PageTitle from "@/components/Common/PageTitle";
-import Pagination from "@/components/Common/Pagination";
 import { CardGridSkeleton } from "@/components/Common/SkeletonLoading";
+import { LocationSearch } from "@/components/Location/LocationSearch";
+
+import useFilters from "@/hooks/useFilters";
 
 import query from "@/Utils/request/query";
 import DeviceCard from "@/pages/Facility/settings/devices/components/DeviceCard";
 import deviceApi from "@/types/device/deviceApi";
+import { LocationList } from "@/types/location/location";
 
 interface Props {
   facilityId: string;
 }
 
 export default function DevicesList({ facilityId }: Props) {
   const { t } = useTranslation();
-  const [page, setPage] = useState(1);
+  const [searchType, setSearchType] = useState<"name" | "location">("name");
+  const [selectedLocation, setSelectedLocation] = useState<LocationList | null>(
+    null,
+  );
 
-  const limit = 12;
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: 12,
+    cacheBlacklist: ["search_text", "current_location"],
+  });
 
   const { data, isLoading } = useQuery({
-    queryKey: ["devices", facilityId, page, limit],
+    queryKey: ["devices", facilityId, qParams, resultsPerPage],
     queryFn: query.debounced(deviceApi.list, {
       pathParams: { facility_id: facilityId },
       queryParams: {
-        offset: (page - 1) * limit,
-        limit,
+        search_text: qParams.search_text,
+        current_location: qParams.current_location,
+        search_type: searchType,
+        limit: resultsPerPage,
+        offset: (qParams.page - 1) * resultsPerPage,
       },
     }),
   });
 
   return (
-    <div className="space-y-6">
-      <div className="flex flex-col sm:flex-row justify-between items-start sm:items-center gap-4">
-        <div className="flex items-center gap-4">
-          <PageTitle title={t("devices")} />
+    <Page title={t("devices")} hideTitleOnPage className="p-0">
+      <div className="space-y-6">
+        <div className="flex flex-col lg:flex-row justify-between gap-4 w-full">
+          <div className="flex flex-col sm:flex-row gap-4">
+            <PageTitle title={t("devices")} />
+          </div>
+          <div className="flex flex-wrap sm:flex-row items-center gap-4">
+            <div className="flex flex-col lg:flex-row items-center gap-2 w-full sm:w-auto">
+              <div className="flex w-full items-center rounded-lg bg-gray-50">
+                <Select
+                  value={searchType}
+                  onValueChange={(value: "name" | "location") => {
+                    setSearchType(value);
+                    updateQuery({
```

Comment:
This results in slightly weird behavior when we switch back and forth.

Searching a name and then switching to location would preserve the name (for this, we will need to add badges to ensure it's clear to user that search is preserved).

However if user were to select a location and then switch back to name again, name would reset.

Vice versa for location as well.